pymc/logprob/rewriting.py

```diff
@@ -351,3 +351,18 @@ def construct_ir_fgraph(
         fgraph.replace_all(new_to_old)
 
     return fgraph, rv_values, memo
+
+
+pregrad_logprob_rewrites_db = SequenceDB()
+pregrad_logprob_rewrites_db.name = "pregrad_logprob_rewrites_db"
+pregrad_logprob_rewrites_db.register("pre-canonicalize", optdb.query("+canonicalize"), "basic")
+pregrad_logprob_rewrites_db.register("stabilize", optdb.query("+stabilize"), "basic")
+
+
+def rewrite_pregrad(graph, rewriter=None):
```

Comment:
I would maybe put this in `pytensorf`, we already have some custom rewrite logic going on there